streams/cluster/src/test/java/com/hortonworks/streamline/streams/cluster/register/impl/ZookeeperServiceRegistrarTest.java

```diff
@@ -50,71 +55,40 @@ public void testRegister() throws Exception {
 
         ZookeeperServiceRegistrar registrar = initializeServiceRegistrar();
 
-        try (InputStream is = getClass().getClassLoader().getResourceAsStream(ZOO_CFG_FILE_PATH)) {
-            Config config = new Config();
-            config.put(ZookeeperServiceRegistrar.PARAM_ZOOKEEPER_SERVER_HOSTNAMES, Lists.newArrayList("zookeeper-1", "zookeeper-2"));
-            ManualServiceRegistrar.ConfigFileInfo zooCfg = new ManualServiceRegistrar.ConfigFileInfo(ZOO_CFG, is);
-            registrar.register(cluster, config, Lists.newArrayList(zooCfg));
-        }
+        Config config = new Config();
+        config.put(ZookeeperServiceRegistrar.PARAM_ZOOKEEPER_SERVER_HOSTNAMES, Lists.newArrayList("zookeeper-1", "zookeeper-2"));
+        config.put(ZookeeperServiceRegistrar.PARAM_ZOOKEEPER_PORT, (Object) 2181);
+        registrar.register(cluster, config, Collections.emptyList());
 
         Service zkService = environmentService.getServiceByName(cluster.getId(), Constants.Zookeeper.SERVICE_NAME);
         assertNotNull(zkService);
-        ServiceConfiguration zooConf = environmentService.getServiceConfigurationByName(zkService.getId(), CONFIGURATION_NAME_ZOO_CFG);
-        assertNotNull(zooConf);
-    }
 
-    @Test
-    public void testRegister_requiredPropertyNotPresent() throws Exception {
-        Cluster cluster = getTestCluster(1L);
-
-        ZookeeperServiceRegistrar registrar = initializeServiceRegistrar();
+        Component zkServer = environmentService.getComponentByName(zkService.getId(), ComponentPropertyPattern.ZOOKEEPER_SERVER.name());
+        assertNotNull(zkServer);
 
-        try (InputStream is = getClass().getClassLoader().getResourceAsStream(ZOO_CFG_BADCASE_FILE_PATH)) {
-            Config config = new Config();
-            config.put(ZookeeperServiceRegistrar.PARAM_ZOOKEEPER_SERVER_HOSTNAMES, Lists.newArrayList("zookeeper-1", "zookeeper-2"));
-            ManualServiceRegistrar.ConfigFileInfo zooCfg = new ManualServiceRegistrar.ConfigFileInfo(ZOO_CFG, is);
-            registrar.register(cluster, config, Lists.newArrayList(zooCfg));
-            fail("Should throw IllegalArgumentException");
-        } catch (IllegalArgumentException e) {
-            // OK
-            Service zkService = environmentService.getServiceByName(cluster.getId(), Constants.Zookeeper.SERVICE_NAME);
-            assertNull(zkService);
-        }
-    }
+        Collection<ComponentProcess> zkServerProcesses = environmentService.listComponentProcessesInComponent(zkServer.getId());
+        assertEquals(Sets.newHashSet("zookeeper-1", "zookeeper-2"),
```

Comment:
Is this always true, or does it depend on the order of the sets? If so it could cause sporadic test failures.